backends/k8s/client.go

```diff
@@ -0,0 +1,219 @@
+
+	tprClient, err := buildTPRClient(config)
+	if err != nil {
+		return nil, err
+	}
+	kubeClient := &Client{
+		clientSet: cs,
+		tprClient: tprClient,
+	}
+
+    	return kubeClient, nil
+}
+
+// Take the etcd like keys and route it to the appropriate k8s API endpoint
+func (c *Client) GetValues(keys []string) (map[string]string, error) {
+	var kvps = make(map[string]string)
+	for _, key := range keys {
+		log.Debug(fmt.Sprintf("Getting key %s", key))
+		if m := singleNode.FindStringSubmatch(key); m != nil {
+			host := m[len(m)-1]
+			node, err := c.clientSet.Nodes().Get(host, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+
+			var prefix = fmt.Sprintf("/calico/bgp/v1/host/%s/", node.Name)
+			for _, ip := range node.Status.Addresses {
+				if ip.Type == kapiv1.NodeInternalIP {
+					kvps[prefix + "ip_addr_v4"] = ip.Address
+					break
+				}
+			}
+
+		}
+		switch key {
+		case global:
+			// Default to "info" until this makes it into k8s
+			kvps[globalLogging] = "info"
+			// Default to 64512
+			kvps[globalASN] = "64512"
+			// Default to False until peering info is available in k8s
+			kvps[globalNodeMesh] = `{"enabled": true}`
+		case globalNodeMesh:
+			// This is needed as there are calls to just 'global' and 'global/loglevel'
+			// Default to False until peering info is available in k8s
+			kvps[globalNodeMesh] = `{"enabled": true}`
+		case ipPool:
+			tprs := thirdparty.IpPoolList{}
+			err := c.tprClient.Get().
+				Resource("ippools").
+				Namespace("kube-system").
+			        Do().Into(&tprs)
+
+			// Ignore not found errors, as this simply means the ippool does
+			// not exist
+			if err != nil {
+				if !kerrors.IsNotFound(err) {
+					return nil, err
+				}
+			}
+
+			for _, pool := range tprs.Items {
+				// We need to parse the JSON value and determine if it is a V4 CIDR
+				// This is ugly, might be better have this included in ippool metadata?
+				vals := []byte(pool.Spec.Value)
+				var d interface{}
+				err = json.Unmarshal(vals, &d)
+				if err != nil {
+					log.Warning("Failed to parse IpPool json")
+					continue
+				}
+				data := d.(map[string]interface{})
+				_, ipNet, err := net.ParseCIDR(data["cidr"].(string))
+				if err != nil {
+					log.Warning(fmt.Sprintf("failed to parse CIDR for %s", pool.Metadata.Name))
+				}
+				if ipNet.Version() == 4 {
+					kvps[ipPool+"/"+pool.Metadata.Name] = pool.Spec.Value
+				}
+			}
+		case allNodes:
+			nodes, err := c.clientSet.Nodes().List(kapiv1.ListOptions{})
+			if err != nil {
+				return nil, err
+			}
+
+			for _, node := range nodes.Items {
+				for _, ip := range node.Status.Addresses {
+					if ip.Type == kapiv1.NodeInternalIP {
+						kvps[allNodes + "/" + node.Name + "/ip_addr_v4"] =  ip.Address
+						break
+					}
+				}
+			}
+		}
+	}
+	fmt.Printf("%v", kvps)
+	return kvps, nil
+}
+
+func (c *Client) WatchPrefix(prefix string, waitIndex uint64, stopChan chan bool, keys []string) (uint64, error) {
```

Comment:
After discussing with @robbrockbank I think we probably do want to implement the watch stuff here.